src/game/pool_game.py

```diff
@@ -1,52 +1,57 @@
+from trueskill import TrueSkill
+
+from src.config import DEFAULT_RANK_ELASTICITY, DEFAULT_RANK_POINTS
 from src.game.game import Game
 
+
 class PoolGame(Game):
     """Represents a game of Pool"""
 
-    @classmethod
-    def add(cls, player_records):
-        """
-        Use a game type and a list of player records to add a PoolGame to db
-        :param player_records a list of PoolPlayerRecords
-        :returns: a game object
-        """
-        game = cls(player_records=[r.key for r in player_records])
-        game.calculate_experience(player_records)
-        game.calculate_rank_points_changes(player_records)
-        game.players = [s.player for s in player_records]
-        game.put()
-        return game  #  TODO: Calculate winner
-
-    def calculate_rank_points_changes(self, player_records):
-        """
-        Calculate points using self.player_records
-        Should be overwritten by subclass
-        """
-        pass
-
-    def calculate_experience(self, player_records):
-        """
-        Use self.player_records to assign experience to the relevant players
-        """
-        pass
-
-
 
 class CutThroatGame(PoolGame):
 
-    def calculate_rank_points_changes(self, player_records):
+    def calculate_rating_change(self):
         """
         Calculate Points using the following formula
         A win is worth 5 points, second place is worth 2 points
         """
-        pass  # TODO
+        player_records = sorted(
+            self.player_records, key=lambda r: r.player_placement)
+        teams = [
+            {r.player.name: r.player.trueskill_rating} for r in player_records
+        ]
+
+        ranks = [r.player_placement for r in player_records]
+
+        env = TrueSkill(
+            mu=DEFAULT_RANK_POINTS, sigma=DEFAULT_RANK_ELASTICITY,
+            draw_probability=0.0)
+        rated_rating_groups = env.rate(teams, ranks=ranks)
+        self._update_ranked_data(player_records, rated_rating_groups)
 
 
 class EightBallGame(PoolGame):
 
-    def calculate_rank_points_changes(self, player_records):
+    def calculate_rating_change(self):
         """
         Calculate Points using the following formula
         A win is worth 3 points
         """
-        pass  # TODO
+        team1 = [r for r in self.player_records if r.player_placement == 1]
+        team2 = [r for r in self.player_records if r.player_placement == 2]
+        player_records = sorted(
+            self.player_records, key=lambda r: r.player_placement)
+        teams = []
+        for team_records in [team1, team2]:
+            team = {}
+            for record in team_records:
+                team[record.player.name] = record.player.trueskill_rating
+            teams.append(team)
```

Comment:
So there is some special logic in place to get team1 and team2. I think this loop makes that logic not very special. Why is this different than just iterating over all the records?
Is it because of the `ranks = [0, 1]`?